.github/workflows/workflow-lint-test-python.yml

```diff
@@ -17,17 +17,18 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install ruff==0.1.0
+        pip install ruff==0.1.0 coverage
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
     - name: Lint with ruff
       run: |
         # stop the build if there are Python syntax errors or undefined names
         ruff --output-format=github --select=E9,F63,F7,F82 --target-version=py311 .
         # default set of ruff rules with GitHub Annotations
         ruff --output-format=github --target-version=py311 .
-    - name: Test with unittest
+    - name: Test and Coverage with unittest
       run: |
-        python -m unittest discover -s tests
+        python -m coverage run -m unittest discover -s tests
+        python -m coverage report --show-missing
```

Comment:
see doc: https://coverage.readthedocs.io/en/7.3.2/cmd.html 

we also want to use --fail-under=X to ensure the step fails if the coverage is not over a specified value by the calling workflow. when the coverage fails, the exit code returned is non-zero and causes the workflow to fail.

example (after tweaking the test)
```
$ python -m coverage report --fail-under=100
Name                 Stmts   Miss  Cover
----------------------------------------
src/__init__.py          0      0   100%
src/main.py              8      3    62%
tests/test_main.py       7      0   100%
----------------------------------------
TOTAL                   15      3    80%
Coverage failure: total of 80 is less than fail-under=100
```

Reply:
yeah I did a fail under when I did it with pytest I didn't add it because you told me I can't fail right away because depending on the dev stage we are developers might not have the whole tests created unless I misunderstood, if that's the case what do you want the threshold to be set as?

Reply:
Set the threshold to a value that represents the current coverage value. 

That way it passes now but we also make sure it doesn't drop below that threshold for new changes.